pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -192,8 +342,12 @@ func (s *Service) marshalAndSend(data interface{}, errorMsg, destination string)
 	return s.outboundTransport.Send(string(jsonString), destination)
 }
 
+func (s *Service) generateRandomID() string {
+	return uuid.New().String()
```

Comment:
not sure why, but the uuid docs says "creates new UUID or panics": https://godoc.org/github.com/google/uuid#New

it's maybe rare, but we should probably not allow panics here.. or at least catch them.

generally speaking, for random bytes, I prefer using crypto.Rand
have you tried  uuid.NewRandom() ?

Reply:
Note uuid.New and uuid.NewRandom are [equivalent](https://github.com/google/uuid/blob/master/version4.go#L14) except for the panic vs error return.


So the following blurb from NewRandom also applies to New.
> The strength of the UUIDs is based on the strength of the crypto/rand

Reply:
@Baha-sk just curious about the error logic for such cases. Whats the error/retry logic processing in such cases (error, panic)? Should we throw error back to the consumers or retry till we succeed?

Reply:
The panic happens if the application [cannot read](https://github.com/google/uuid/blob/master/version4.go#L37) from crypto/rand. This seems a bit unrecoverable...